classes/UpgradeTools/Module/ModuleMigration.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\UpgradeTools\Module;
+
+use LogicException;
+use PrestaShop\Module\AutoUpgrade\Exceptions\UpgradeException;
+use PrestaShop\Module\AutoUpgrade\Log\Logger;
+use PrestaShop\Module\AutoUpgrade\UpgradeTools\Translator;
+
+class ModuleMigration
+{
+    /** @var Translator */
+    private $translator;
+
+    /** @var Logger */
+    private $logger;
+
+    /** @var string|null */
+    private $module_name;
+
+    /** @var string|null */
+    private $db_version;
+
+    /** @var string|null */
+    private $local_version;
+
+    /** @var string[]|null */
+    private $migration_files;
+
+    /** @var string|null */
+    private $upgrade_files_root_path;
+
+    /** @var \Module|null */
+    private $module_instance;
+
+    public function __construct(Translator $translator, Logger $logger)
+    {
+        $this->translator = $translator;
+        $this->logger = $logger;
+        $this->module_name = null;
+        $this->db_version = null;
+        $this->local_version = null;
+        $this->migration_files = null;
+        $this->upgrade_files_root_path = null;
+        $this->module_instance = null;
```

Comment:
Prefer using variable names in camel case

```suggestion
        $this->moduleName = null;
        $this->dbVersion = null;
        $this->localVersion = null;
        $this->migrationFiles = null;
        $this->upgradeFilesRootPath = null;
        $this->moduleInstance = null;
```